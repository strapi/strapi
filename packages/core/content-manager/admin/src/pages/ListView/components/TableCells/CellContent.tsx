import { Tooltip, Typography } from '@strapi/design-system';
import isEmpty from 'lodash/isEmpty';

import { CellValue } from './CellValue';
import { SingleComponent, RepeatableComponent } from './Components';
import { MediaSingle, MediaMultiple } from './Media';
import { RelationMultiple, RelationSingle } from './Relations';

import type { ListFieldLayout } from '../../../../hooks/useDocumentLayout';
import type { Schema, Data } from '@strapi/types';

interface CellContentProps extends Omit<ListFieldLayout, 'cellFormatter'> {
  content: Schema.Attribute.Value<Schema.Attribute.AnyAttribute>;
  rowId: Data.ID;
}

const CellContent = ({ content, mainField, attribute, rowId, name }: CellContentProps) => {
  if (!hasContent(content, mainField, attribute)) {
    return (
      <Typography
        textColor="neutral800"
        paddingLeft={attribute.type === ('relation' || 'component') ? '1.6rem' : 0}
        paddingRight={attribute.type === ('relation' || 'component') ? '1.6rem' : 0}
      >
        -
      </Typography>
    );
  }

  switch (attribute.type) {
    case 'media':
      if (!attribute.multiple) {
        return <MediaSingle {...content} />;
      }

      return <MediaMultiple content={content} />;

    case 'relation': {
      if (isSingleRelation(attribute.relation)) {
        return <RelationSingle mainField={mainField} content={content} />;
      }

      return <RelationMultiple rowId={rowId} mainField={mainField} content={content} name={name} />;
    }

    case 'component':
      if (attribute.repeatable) {
        return <RepeatableComponent mainField={mainField} content={content} />;
      }

      return <SingleComponent mainField={mainField} content={content} />;

    case 'string':
      return (
        <Tooltip description={content}>
          <Typography maxWidth="30rem" ellipsis textColor="neutral800">
            <CellValue type={attribute.type} value={content} />
          </Typography>
        </Tooltip>
      );

    default:
      return (
        <Typography maxWidth="30rem" ellipsis textColor="neutral800">
          <CellValue type={attribute.type} value={content} />
        </Typography>
      );
  }
};

const hasContent = (
  content: CellContentProps['content'],
  mainField: CellContentProps['mainField'],
  attribute: CellContentProps['attribute']
) => {
  if (attribute.type === 'component') {
    // Repeatable fields show the ID as fallback, in case the mainField
    // doesn't have any content
    if (attribute.repeatable || !mainField) {
      return content?.length > 0;
    }

    const value = content?.[mainField.name];

    // relations, media ... show the id as fallback
    if (mainField.name === 'id' && ![undefined, null].includes(value)) {
      return true;
    }

    return !isEmpty(value);
  }

  if (attribute.type === 'relation') {
    if (isSingleRelation(attribute.relation)) {
      return !isEmpty(content);
    }

    if (Array.isArray(content)) {
      return content.length > 0;
    }

    return content?.count > 0;
  }

  /*
      Biginteger fields need to be treated as strings, as `isNumber`
      doesn't deal with them.
  */
  if (['integer', 'decimal', 'float', 'number'].includes(attribute.type)) {
    return typeof content === 'number';
  }

  if (attribute.type === 'boolean') {
    return content !== null;
  }

  return !isEmpty(content);
};

const isSingleRelation = (
  type: Extract<CellContentProps['attribute'], { type: 'relation' }>['relation']
) => ['oneToOne', 'manyToOne', 'oneToOneMorph'].includes(type);

export { CellContent };
export type { CellContentProps };
