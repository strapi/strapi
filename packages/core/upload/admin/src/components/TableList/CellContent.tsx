import { Typography } from '@strapi/design-system';
import parseISO from 'date-fns/parseISO';
import { useIntl } from 'react-intl';

import { formatBytes, getFileExtension } from '../../utils';

import { PreviewCell } from './PreviewCell';

import type { File } from '../../../../shared/contracts/files';

export interface CellContentProps {
  cellType: string;
  contentType?: string;
  content: File;
  name: string;
}

export const CellContent = ({ cellType, contentType, content, name }: CellContentProps) => {
  const { formatDate, formatMessage } = useIntl();
  const contentValue = content[name as Extract<keyof File, string>];

  switch (cellType) {
    case 'image':
      return <PreviewCell type={contentType} content={content} />;

    case 'date':
      if (typeof contentValue === 'string') {
        return <Typography>{formatDate(parseISO(contentValue), { dateStyle: 'full' })}</Typography>;
      }

    case 'size':
      if (contentType === 'folder')
        return (
          <Typography
            aria-label={formatMessage({
              id: 'list.table.content.empty-label',
              defaultMessage: 'This field is empty',
            })}
          >
            -
          </Typography>
        );
      if (typeof contentValue === 'string' || typeof contentValue === 'number') {
        return <Typography>{formatBytes(contentValue)}</Typography>;
      }

    case 'ext':
      if (contentType === 'folder')
        return (
          <Typography
            aria-label={formatMessage({
              id: 'list.table.content.empty-label',
              defaultMessage: 'This field is empty',
            })}
          >
            -
          </Typography>
        );

      if (typeof contentValue === 'string') {
        return <Typography>{getFileExtension(contentValue)?.toUpperCase()}</Typography>;
      }
    case 'text':
      if (typeof contentValue === 'string') {
        return <Typography>{contentValue}</Typography>;
      }

    default:
      return (
        <Typography
          aria-label={formatMessage({
            id: 'list.table.content.empty-label',
            defaultMessage: 'This field is empty',
          })}
        >
          -
        </Typography>
      );
  }
};
