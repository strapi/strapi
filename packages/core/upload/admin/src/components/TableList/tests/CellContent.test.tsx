import { DesignSystemProvider } from '@strapi/design-system';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import { CellContent, CellContentProps } from '../CellContent';

const PROPS_FIXTURE = {
  cellType: 'image',
  contentType: 'asset',
  content: {
    id: 1,
    name: 'michka-picture-name',
    hash: 'michka-picture-hash',
    alternativeText: 'alternative alt',
    ext: 'jpeg',
    formats: {
      thumbnail: {
        url: 'michka-picture-url-thumbnail.jpeg',
      },
    },
    mime: 'image/jpeg',
    url: 'michka-picture-url-default.jpeg',
  },
  name: 'preview',
};

const ComponentFixture = (props: Partial<CellContentProps>) => {
  const customProps = {
    ...PROPS_FIXTURE,
    ...props,
  };

  return (
    <IntlProvider locale="en" messages={{}}>
      <DesignSystemProvider>
        <CellContent {...PROPS_FIXTURE} {...customProps} />
      </DesignSystemProvider>
    </IntlProvider>
  );
};

const setup = (props: Partial<CellContentProps>) => render(<ComponentFixture {...props} />);

describe('TableList | CellContent', () => {
  const fileTypesToTest = [
    { mime: 'application/pdf', ext: 'pdf', expectedIcon: 'file-pdf-icon' },
    { mime: 'application/vnd.ms-excel', ext: 'xls', expectedIcon: 'file-xls-icon' },
    { mime: 'text/csv', ext: 'csv', expectedIcon: 'file-csv-icon' },
    { mime: 'application/zip', ext: 'zip', expectedIcon: 'file-zip-icon' },
    { mime: 'text/plain', ext: 'txt', expectedIcon: 'file-icon' },
  ];
  fileTypesToTest.forEach((fileType) => {
    it(`should render the ${fileType.expectedIcon === 'file-icon' ? 'default file' : 'corresponding'} icon according to the file type (${fileType.ext})`, () => {
      const { ext, expectedIcon, mime } = fileType;
      const { container, getByTestId } = setup({
        content: { ...PROPS_FIXTURE.content, mime, ext },
      });

      expect(getByTestId(expectedIcon)).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });

  it('should render image cell type when element type is folder', () => {
    const { container, getByLabelText } = setup({ contentType: 'folder' });

    expect(getByLabelText('folder')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render text cell type', () => {
    const { container, getByText } = setup({
      cellType: 'text',
      content: { ...PROPS_FIXTURE.content, name: 'some text' },
      name: 'name',
    });

    expect(getByText('some text')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render extension cell type when element type is asset', () => {
    const { container, getByText } = setup({
      cellType: 'ext',
      content: { ...PROPS_FIXTURE.content, ext: '.pdf' },
      name: 'ext',
    });

    expect(getByText('PDF')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render extension cell type with "-" when element type is folder', () => {
    const { container, getByText } = setup({ cellType: 'ext', contentType: 'folder' });

    expect(getByText('-')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render size cell type when element type is asset', () => {
    const { container, getByText } = setup({
      cellType: 'size',
      content: { ...PROPS_FIXTURE.content, size: Number('20.5435') },
      name: 'size',
    });

    expect(getByText('21KB')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render size cell type with "-" when element type is folder', () => {
    const { container, getByText } = setup({ cellType: 'size', contentType: 'folder' });

    expect(getByText('-')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render date cell type', () => {
    const { container, getByText } = setup({
      cellType: 'date',
      content: { ...PROPS_FIXTURE.content, updatedAt: '2022-11-18T12:08:02.202Z' },
      name: 'updatedAt',
    });

    expect(getByText('Friday, November 18, 2022')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render "-" by default when no recognized cell type is passed', () => {
    const { container, getByText } = setup({ cellType: 'not recognized type' });

    expect(getByText('-')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
