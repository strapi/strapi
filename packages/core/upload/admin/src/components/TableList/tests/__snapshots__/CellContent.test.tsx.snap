// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableList | CellContent should render "-" by default when no recognized cell type is passed 1`] = `
.c0 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: currentcolor;
}

.c1 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <span
    aria-label="This field is empty"
    class="c0"
  >
    -
  </span>
  <span
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render date cell type 1`] = `
.c0 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: currentcolor;
}

.c1 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <span
    class="c0"
  >
    Friday, November 18, 2022
  </span>
  <span
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render extension cell type when element type is asset 1`] = `
.c0 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: currentcolor;
}

.c1 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <span
    class="c0"
  >
    PDF
  </span>
  <span
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render extension cell type with "-" when element type is folder 1`] = `
.c0 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: currentcolor;
}

.c1 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <span
    aria-label="This field is empty"
    class="c0"
  >
    -
  </span>
  <span
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render image cell type when element type is folder 1`] = `
.c0 {
  width: 3.2rem;
  height: 3.2rem;
  background: #eaf5ff;
  border-radius: 50%;
}

.c1 {
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: flex;
}

.c2 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <div
    class="c0 c1"
  >
    <svg
      aria-label="folder"
      fill="#66b7f1"
      height="1.6rem"
      viewBox="0 0 32 32"
      width="1.6rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M27 9H16.414L13 5.586A1.98 1.98 0 0 0 11.586 5H5a2 2 0 0 0-2 2v18.078A1.926 1.926 0 0 0 4.924 27H27.11A1.89 1.89 0 0 0 29 25.111V11a2 2 0 0 0-2-2M5 7h6.586l2 2H5z"
      />
    </svg>
  </div>
  <span
    class="c2"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render size cell type when element type is asset 1`] = `
.c0 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: currentcolor;
}

.c1 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <span
    class="c0"
  >
    21KB
  </span>
  <span
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render size cell type with "-" when element type is folder 1`] = `
.c0 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: currentcolor;
}

.c1 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <span
    aria-label="This field is empty"
    class="c0"
  >
    -
  </span>
  <span
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render text cell type 1`] = `
.c0 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: currentcolor;
}

.c1 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <span
    class="c0"
  >
    some text
  </span>
  <span
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render the corresponding icon according to the file type (csv) 1`] = `
.c0 {
  width: 3.2rem;
  height: 3.2rem;
  color: #8e8ea9;
  background: #f6f6f9;
  border-radius: 100%;
}

.c1 {
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: flex;
}

.c2 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <div
    class="c0 c1"
  >
    <svg
      data-testid="file-csv-icon"
      fill="currentColor"
      height="16"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m213.66 82.34-56-56A8 8 0 0 0 152 24H56a16 16 0 0 0-16 16v76a4 4 0 0 0 4 4h168a4 4 0 0 0 4-4V88a8 8 0 0 0-2.34-5.66M152 88V44l44 44ZM48 180c0 11 7.18 20 16 20a14.18 14.18 0 0 0 10.06-4.5 8.21 8.21 0 0 1 10.9-.91 8 8 0 0 1 .82 11.81A30.06 30.06 0 0 1 64 216c-17.64 0-32-16.15-32-36s14.36-36 32-36a30 30 0 0 1 21.39 9.19 8.26 8.26 0 0 1 .73 11.09 8 8 0 0 1-11.9.38A14.17 14.17 0 0 0 64 160c-8.82 0-16 9-16 20m103.81 16.31a20.82 20.82 0 0 1-9.19 15.23C137.43 215 131 216 125.13 216a61.1 61.1 0 0 1-15.13-2 8 8 0 1 1 4.3-15.41c4.38 1.2 14.95 2.7 19.55-.36.88-.59 1.83-1.52 2.14-3.93.35-2.67-.71-4.1-12.78-7.59-9.35-2.7-25-7.23-23-23.11a20.56 20.56 0 0 1 9-14.95c11.84-8 30.71-3.31 32.83-2.76a8 8 0 0 1-4.07 15.48c-4.49-1.17-15.23-2.56-19.83.56a4.54 4.54 0 0 0-2 3.67c-.12.9-.14 1.09 1.11 1.9 2.31 1.49 6.45 2.68 10.45 3.84 9.79 2.83 26.35 7.66 24.11 24.97M215.42 155l-19.89 55.68a8 8 0 0 1-15.06 0L160.58 155a8.21 8.21 0 0 1 4.5-10.45 8 8 0 0 1 10.45 4.76l12.47 34.9 12.47-34.9a8 8 0 0 1 10.45-4.76 8.23 8.23 0 0 1 4.5 10.45"
      />
    </svg>
  </div>
  <span
    class="c2"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render the corresponding icon according to the file type (pdf) 1`] = `
.c0 {
  width: 3.2rem;
  height: 3.2rem;
  color: #8e8ea9;
  background: #f6f6f9;
  border-radius: 100%;
}

.c1 {
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: flex;
}

.c2 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <div
    class="c0 c1"
  >
    <svg
      data-testid="file-pdf-icon"
      fill="currentColor"
      height="16"
      viewBox="0 0 32 32"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 15h20a1 1 0 0 0 1-1v-3a1 1 0 0 0-.293-.707l-7-7A1 1 0 0 0 19 3H7a2 2 0 0 0-2 2v9a1 1 0 0 0 1 1m13-9.5 5.5 5.5H19zM28 19a1 1 0 0 1-1 1h-3v2h2a1 1 0 0 1 0 2h-2v2a1 1 0 0 1-2 0v-7a1 1 0 0 1 1-1h4a1 1 0 0 1 1 1M8 18H6a1 1 0 0 0-1 1v7a1 1 0 1 0 2 0v-1h1a3.5 3.5 0 1 0 0-7m0 5H7v-3h1a1.5 1.5 0 1 1 0 3m8-5h-2a1 1 0 0 0-1 1v7a1 1 0 0 0 1 1h2a4.5 4.5 0 1 0 0-9m0 7h-1v-5h1a2.5 2.5 0 0 1 0 5"
      />
    </svg>
  </div>
  <span
    class="c2"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render the corresponding icon according to the file type (xls) 1`] = `
.c0 {
  width: 3.2rem;
  height: 3.2rem;
  color: #8e8ea9;
  background: #f6f6f9;
  border-radius: 100%;
}

.c1 {
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: flex;
}

.c2 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <div
    class="c0 c1"
  >
    <svg
      data-testid="file-xls-icon"
      fill="currentColor"
      height="16"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M44 120h168a4 4 0 0 0 4-4V88a8 8 0 0 0-2.34-5.66l-56-56A8 8 0 0 0 152 24H56a16 16 0 0 0-16 16v76a4 4 0 0 0 4 4m108-76 44 44h-44Zm4 164.53a8.18 8.18 0 0 1-8.25 7.47H120a8 8 0 0 1-8-8v-55.73a8.18 8.18 0 0 1 7.47-8.25 8 8 0 0 1 8.53 8v48h20a8 8 0 0 1 8 8.51m-61.49-51.88L77.83 180l16.68 23.35a8 8 0 0 1-13 9.3L68 193.76l-13.49 18.89a8 8 0 1 1-13-9.3L58.17 180l-16.68-23.35a8 8 0 0 1 2.3-11.46 8.19 8.19 0 0 1 10.88 2.38L68 166.24l13.49-18.89a8 8 0 0 1 13 9.3Zm121.28 39.66a20.81 20.81 0 0 1-9.18 15.23c-5.19 3.46-11.67 4.46-17.49 4.46a60.6 60.6 0 0 1-15.19-2 8 8 0 0 1 4.31-15.41c4.38 1.21 14.94 2.71 19.54-.35.89-.6 1.84-1.52 2.15-3.93.34-2.67-.72-4.1-12.78-7.59-9.35-2.7-25-7.23-23-23.12a20.58 20.58 0 0 1 8.95-14.94c11.84-8 30.72-3.31 32.83-2.76a8 8 0 0 1-4.07 15.48c-4.48-1.17-15.22-2.56-19.82.56a4.54 4.54 0 0 0-2 3.67c-.11.9-.13 1.08 1.12 1.9 2.31 1.49 6.45 2.68 10.45 3.84 9.87 2.82 26.39 7.65 24.18 24.96"
      />
    </svg>
  </div>
  <span
    class="c2"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render the corresponding icon according to the file type (zip) 1`] = `
.c0 {
  width: 3.2rem;
  height: 3.2rem;
  color: #8e8ea9;
  background: #f6f6f9;
  border-radius: 100%;
}

.c1 {
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: flex;
}

.c2 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <div
    class="c0 c1"
  >
    <svg
      data-testid="file-zip-icon"
      fill="currentColor"
      height="16"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M184 144h-16a8 8 0 0 0-8 8v55.73a8.17 8.17 0 0 0 7.47 8.25 8 8 0 0 0 8.53-8v-8h7.4c15.24 0 28.14-11.92 28.59-27.15A28 28 0 0 0 184 144m-.35 40H176v-24h8a12 12 0 0 1 12 13.16A12.25 12.25 0 0 1 183.65 184M136 152v55.73a8.17 8.17 0 0 1-7.47 8.25 8 8 0 0 1-8.53-8v-55.71a8.17 8.17 0 0 1 7.47-8.25A8 8 0 0 1 136 152m-40 56.53a8.17 8.17 0 0 1-8.27 7.47h-31.5a8.27 8.27 0 0 1-6-2.5 8 8 0 0 1-1.18-9.5l25.16-44H56.27a8.17 8.17 0 0 1-8.27-7.47 8 8 0 0 1 8-8.53h31.77a8.27 8.27 0 0 1 6 2.5A8 8 0 0 1 95 156l-25.21 44H88a8 8 0 0 1 8 8.53M213.66 82.34l-56-56A8 8 0 0 0 152 24H56a16 16 0 0 0-16 16v76a4 4 0 0 0 4 4h168a4 4 0 0 0 4-4V88a8 8 0 0 0-2.34-5.66M152 88V44l44 44Z"
      />
    </svg>
  </div>
  <span
    class="c2"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;

exports[`TableList | CellContent should render the default file icon according to the file type (txt) 1`] = `
.c0 {
  width: 3.2rem;
  height: 3.2rem;
  color: #8e8ea9;
  background: #f6f6f9;
  border-radius: 100%;
}

.c1 {
  align-items: center;
  justify-content: center;
  flex-direction: row;
  display: flex;
}

.c2 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
  <div
    class="c0 c1"
  >
    <svg
      data-testid="file-icon"
      fill="currentColor"
      height="16"
      viewBox="0 0 32 32"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m26.708 10.293-7-7A1 1 0 0 0 19 3H7a2 2 0 0 0-2 2v22a2 2 0 0 0 2 2h18a2 2 0 0 0 2-2V11a1 1 0 0 0-.293-.707M19 11V5.5l5.5 5.5z"
      />
    </svg>
  </div>
  <span
    class="c2"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;
